tests/assembly/cmse.rs

```diff
@@ -10,14 +13,85 @@ pub trait Sized {}
 pub trait Copy {}
 
 // CHECK-LABEL: __acle_se_entry_point
-// CHECK: bxns
+// CHECK: entry_point:
+//
+// Write return argument (two registers since 64bit integer)
+// CHECK: movs r0, #0
+// CHECK: movs r1, #0
+//
+// If we are using hard-float:
+// * Check if the float registers were touched (bit 3 in CONTROL)
+// hard: mrs     r12, control
+// hard: tst.w   r12, #8
+// hard: beq     .LBB0_2
+//
+// * If touched clear all float registers (d0..=d7)
+// hard: vmov    d0, lr, lr
+// hard: vmov    d1, lr, lr
+// hard: vmov    d2, lr, lr
+// hard: vmov    d3, lr, lr
+// hard: vmov    d4, lr, lr
+// hard: vmov    d5, lr, lr
+// hard: vmov    d6, lr, lr
+// hard: vmov    d7, lr, lr
+//
+// * If touched clear FPU status register
+// hard: vmrs    r12, fpscr
+// hard: bic     r12, r12, #159
+// hard: bic     r12, r12, #4026531840
+// hard: vmsr    fpscr, r12
+// hard: .LBB0_2:
+//
+// Clear all other registers that might have been used
+// CHECK: mov r2, lr
+// CHECK: mov r3, lr
+// CHECK: mov r12, lr
+//
+// Clear the flags
+// CHECK: msr apsr_nzcvq, lr
+//
+// Branch back to non-secure side
+// CHECK: bxns lr
 #[no_mangle]
 pub extern "C-cmse-nonsecure-entry" fn entry_point() -> i64 {
     0
 }
 
+// NOTE for future codegen changes:
+// The specific register assignment is not important, however:
```

Comment:
Problem: if it's not important which register gets assigned, then please capture the register used with something like `[[r[0-9]+:REG1]]` or something (I don't recall the exact `FileCheck` syntax).

Reply:
I guess this could even just leave out the second operand, I'll try that!

Reply:
Leaving if the operands worked, makes it even less noisy and just keeps the relevant parts.

Reply:
Problem: see above review comment, it's fine to leave out register assignments that aren't related to clearing, but we should make sure if a register serves as a zero register used for clearing then that is matched on.